oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunctionITests.java

```diff
@@ -0,0 +1,269 @@
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.authentication.TestingAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.InMemoryOAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.registration.TestClientRegistrations;
+import org.springframework.security.oauth2.client.web.AuthenticatedPrincipalOAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.OAuth2RefreshToken;
+import org.springframework.security.oauth2.core.TestOAuth2RefreshTokens;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.blockhound.BlockHound;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.HashSet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+import static org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction.clientRegistrationId;
+
+/**
+ * @author Joe Grandja
+ */
+public class ServletOAuth2AuthorizedClientExchangeFilterFunctionITests {
+	private ClientRegistrationRepository clientRegistrationRepository;
+	private OAuth2AuthorizedClientRepository authorizedClientRepository;
+	private ServletOAuth2AuthorizedClientExchangeFilterFunction authorizedClientFilter;
+	private MockWebServer server;
+	private String serverUrl;
+	private WebClient webClient;
+	private Authentication authentication;
+	private MockHttpServletRequest request;
+	private MockHttpServletResponse response;
+
+	@BeforeClass
+	public static void setUpBlockingChecks() {
+		// IMPORTANT:
+		// Before enabling BlockHound, we need to force the initialization of
+		// java.lang.Package.defineSystemPackage(). When the JVM loads java.lang.Package.getSystemPackage(),
+		// it attempts to java.lang.Package.loadManifest() which is blocking I/O and triggers BlockHound to error.
+		// The following code forces the loading of the manifest.
+		// NOTE: This is an issue with JDK 8. It's been tested on JDK 10 and works fine w/o this workaround.
+		Class.class.getPackage();
```

Comment:
you could also whitelist it:
https://github.com/reactor/BlockHound/blob/1.0.0.M4/docs/customization.md#dis-allowing-blocking-calls-inside-methods

Reply:
@bsideup It does block, so why would we whitelist it?